storage-access.bs

```diff
@@ -162,7 +162,15 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>ha
 
     ISSUE: "same authority" here is a placeholder for a future concept that allows user agents to perform [=same site=] checks while adhering to additional security aspects such as the presence of a cross-site parent document, see [whatwg/storage#142](https://github.com/whatwg/storage/issues/142#issuecomment-1122147159). In practice, this might involve comparing the [=site for cookies=] or performing a [=same site=] check with the top-level document.
 
-1. [=Queue a global task=] on the [=permissions task source=] given |global| to [=/resolve=] |p| with |global|'s [=environment/has storage access=].
+1. Run the following steps [=in parallel=]:
+    1. Let |Whether the User Agent Allows Unpartitioned Cookie Access| be an algorithm that, given a user agent's settings object |user agent settings|, runs the following steps:
+        1. [=Queue a global task=] on the [=permission task source=] given |global| to:
+            1. If |global|'s [=environment/has storage access=] is false:
+                1. If |user agent settings| allows unpartitioned cookie access, [=/resolve=] |p| with true and return |p|.
```

Comment:
As a drive-by note, I think it would be nice to use something like `if the user agent's settings explicitly allow unpartitioned cookie access for |doc|` (emphasis on the *explicit* here) and then have a non-normative note informing where such an explicit allow could come from (i.e. the user changing global browser settings or per-site allow-lists).

Reply:
Should it be _doc_ or some kind of (site, site) tuple? With the former it doesn't seem like the kind of thing you'd want to do in parallel.

Reply:
I think I asked you this before but are there rules around which concepts we can reference in parallel vs. not? It feels like |doc| may be enough information to allow the implementer to infer that it should be whatever reference to the doc the main process holds, but OTOH it's also easy to extract the (site, site) pair from the doc ahead of time so this is more of a principle question :)

Reply:
No exact rules. But making things somewhat explicit is generally preferred.

Reply:
Thanks for the input! Is there any existing `(site, site) tuple` use case I can take a look to get some idea of the writing format?

Reply:
It's what we currently use as permission key type: https://privacycg.github.io/storage-access/#permissions-integration.

Reply:
Still looking into how to rephrase with mentioning `(site, site) tuple` instead of `doc`. But could you PTAL the current steps first? Thanks!